### PR TITLE
Feature: Project group dropdown for Octopus Deploy

### PR DESCRIPTION
**Pull request by NiekRossenInfoSupport**
## Hey, I just made a Pull Request!

     
This PR allows developers to use a dropdown for selecting Octopus Project Groups. It will contain all the names of the project groups. The user can pick 1 project group as the selected one. If no project groups are found, the user can provide a project group by using an input field.

The dropdown will look as follows
image

The dropdown can be implemented using the following `template.yaml`
```yaml
apiVersion: scaffolder.backstage.io/v1beta3
kind: Template
metadata:
  ...
spec:
  ...
  parameters:
    - title: Octopus Project Name
      properties:
        projectName:
          title: Octopus Project Name
          type: string
          ui:field: OctopusDeployProjectGroupDropdown
```

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/octopus-deploy/CHANGELOG.md
Comment (tudi2d) [inline plugins/octopus-deploy/CHANGELOG.md:9]: Make sure to not direclty adjust the Changelog, but run `yarn changeset` to create a changeset instead, which will than automatically updated the Changelog once approved:)

File: plugins/octopus-deploy/README.md
Comment (tudi2d) [inline plugins/octopus-deploy/README.md:47]: Maybe remove the `1.` in line `26` when you no longer need that enummeration

File: plugins/octopus-deploy/package.json
Comment (tudi2d) [inline plugins/octopus-deploy/package.json:3]: no need to manually update this as well.

File: plugins/octopus-deploy/src/api/index.ts
Comment (tudi2d) [inline plugins/octopus-deploy/src/api/index.ts:170]: Wondering if it might make sense to abstract this into a private function similar to `getProjectApiUrl`, but for project groups (or adjust the current abstraction to support project groups urls).

File: plugins/octopus-deploy/src/components/ScaffolderDropdown/ProjectGroupDropdown.tsx
Comment (tudi2d) [inline plugins/octopus-deploy/src/components/ScaffolderDropdown/ProjectGroupDropdown.tsx:19]: This imports can be combined

File: plugins/octopus-deploy/src/hooks/useProjectGroups.ts
Comment (tudi2d) [inline plugins/octopus-deploy/src/hooks/useProjectGroups.ts:32]: you could ensure that this defaults to `[]`, when `value` is not defined to avoid the other `undefined` in the components.

File: plugins/octopus-deploy/src/components/ScaffolderDropdown/ProjectGroupDropdown.tsx
Comment (tudi2d) [inline plugins/octopus-deploy/src/components/ScaffolderDropdown/ProjectGroupDropdown.tsx:53]: Can you please explain this line? Why only `p[0]`? Why would a single project group be an array?

**Review by tudi2d — CHANGES_REQUESTED** [review]
Thanks for your contribution & congrats to your first PR on Backstage! First review before someone from the scaffolder maintainers looks at it. Please take a look at the comment mentioning the missing changeset & maybe read how we create our changelog.

**Comment by NiekRossenInfoSupport** [comment]
Hi i need some help regarding the ProjectGroupDropdownProps, at the moment im using `makeFieldSchemaFromZod` from `@backstage/plugin-scaffolder`. This will make the CI/CD fail. Is there any way this can be implemented in another way?

File: plugins/octopus-deploy/src/api/index.ts
Comment (NiekRossenInfoSupport) [reply plugins/octopus-deploy/src/api/index.ts]: I've updated the implementation a bit. Maybe later we can look at using the Typescript Octopus Api package created by octopus itself [here](https://github.com/OctopusDeploy/api-client.ts)

File: plugins/octopus-deploy/src/components/ScaffolderDropdown/ProjectGroupDropdown.tsx
Comment (benjdlambert) [inline plugins/octopus-deploy/src/components/ScaffolderDropdown/ProjectGroupDropdown.tsx:48]: There's a `ScaffolderField` component that you can use here to refactor away some of the boilerplate for a Field Extension.

https://github.com/backstage/backstage/blob/d202a258d181e0ff7e726de30e105107ca9e4647/plugins/scaffolder-react/src/next/components/ScaffolderField/ScaffolderField.tsx#L53

Should make this a little cleaner.

**Review by benjdlambert — CHANGES_REQUESTED** [review]
This is looking pretty good, just a small nit about using `import { ScaffolderField } from '@backstage/plugin-scaffolder-react` instead :pray:

File: plugins/octopus-deploy/src/components/ScaffolderDropdown/ProjectGroupDropdown.tsx
Comment (NiekRossenInfoSupport) [reply plugins/octopus-deploy/src/components/ScaffolderDropdown/ProjectGroupDropdown.tsx]: Refactored it to use the ScaffolderField now :)

**Comment by tudi2d** [comment]
heads-up that API reports are failing:)

**Comment by benjdlambert** [comment]
@NiekRossenInfoSupport think this just needs a `yarn install` and commit the `yarn.lock` changes :pray:

**Comment by NiekRossenInfoSupport** [comment]
Thanks for the help! I was being a bit sloppy :P

**Comment by benjdlambert** [comment]
@NiekRossenInfoSupport no stress :pray:

**Review by benjdlambert — APPROVED** [review]

**Comment by NiekRossenInfoSupport** [comment]
Thanks! @benjdlambert , do you have any idea when the next release is and when this will be available?

**Comment by benjdlambert** [comment]
@NiekRossenInfoSupport the next release is coming momentarily. :joy: It's why i'm trying to get this through. 1.24 comes out today when we press release 🤞

**Comment by NiekRossenInfoSupport** [comment]
Oke nice! I appreciate it 😄👍

**Pull request merged**
